python/tests/test_pycoro.py

```diff
@@ -0,0 +1,152 @@
+# SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+
+import pytest
+
+from mrc._pymrc.tests.coro.coro import call_async
+from mrc._pymrc.tests.coro.coro import call_fib_async
+from mrc._pymrc.tests.coro.coro import raise_at_depth_async
+from mrc.core import coro as pycoro
+
+
+@pytest.mark.asyncio
+async def test_coro():
+
+    # hit_inside = False
+
+    async def inner():
+
+        # nonlocal hit_inside
+
+        result = await pycoro.wrap_coroutine(asyncio.sleep(1, result=['a', 'b', 'c']))
+
+        # hit_inside = True
+
+        return [result]
+
+    returned_val = await pycoro.wrap_coroutine(inner())
+
+    assert returned_val == 'a'
+    # assert hit_inside
+
+
+@pytest.mark.asyncio
+async def test_coro_many():
+
+    expected_count = 1000
+    # hit_count = 0
+
+    start_time = asyncio.get_running_loop().time()
+
+    async def inner():
+
+        # nonlocal hit_count
+
+        await asyncio.sleep(0.1)
+
+        # hit_count += 1
+
+        return ['a', 'b', 'c']
+
+    coros = [pycoro.wrap_coroutine(inner()) for _ in range(expected_count)]
+
+    returned_vals = await asyncio.gather(*coros)
+
+    end_time = asyncio.get_running_loop().time()
+
+    assert returned_vals == ['a'] * expected_count
+    # assert hit_count == expected_count
```

Comment:
Are these going to stay or be expanded on or can we remove?

Reply:
@drobison00 yeah the only way to do that is with the CoroutineRunnable. That's a separate issue and the next PR I'm working on.

Reply:
ah good catch, thanks. I'll add these back in. Had to comment them out while I was tracking down a bug in my test code.